pkg/pull/pull.go

```diff
@@ -604,6 +602,31 @@ func Pull(upstreamURI string, pullOptions PullOptions) (string, error) {
 		return "", errors.Wrap(err, "failed to write rendered")
 	}
 
+	// preflights may also be included within helm chart templates, so load any from the rendered dir
+	tsKinds, err := kotsutil.LoadTSKindsFromPath(u.GetRenderedDir(writeUpstreamOptions))
+	if err != nil {
+		return "", errors.Wrap(err, fmt.Sprintf("failed to load troubleshoot kinds from path: %s", u.GetRenderedDir(writeUpstreamOptions)))
+	}
+
+	if tsKinds.PreflightsV1Beta2 != nil {
+		var renderedPreflight *troubleshootv1beta2.Preflight
+		for _, v := range tsKinds.PreflightsV1Beta2 {
+			renderedPreflight = troubleshootpreflight.ConcatPreflightSpec(renderedPreflight, &v)
+		}
+
+		if renderedPreflight != nil {
+			renderedPreflightBytes, err := kotsutil.MarshalRuntimeObject(renderedPreflight)
+			if err != nil {
+				return "", errors.Wrap(err, "failed to marshal rendered preflight")
+			}
+			renderedKotsKindsMap["helm-preflight.yaml"] = renderedPreflightBytes
+		}
+	}
+
+	if err := kotsutil.WriteKotsKinds(renderedKotsKindsMap, u.GetKotsKindsDir(writeUpstreamOptions)); err != nil {
```

Comment:
nit pick, can you move this code block to be right before this line? just want to make sure the installation object gets updated with the latest information immediately before it's written to `kotsKinds`, that way it's less prone to error if something updates the installation object but forgets to update it here.
```
// installation spec gets updated during the process, ensure the map has the latest version
	installationBytes, err := os.ReadFile(filepath.Join(u.GetUpstreamDir(writeUpstreamOptions), "userdata", "installation.yaml"))
if err != nil {
	return "", errors.Wrap(err, "failed to read installation file")
}
renderedKotsKindsMap["userdata/installation.yaml"] = []byte(installationBytes)
```